cpp/arcticdb/stream/append_map.cpp

```diff
@@ -213,31 +213,53 @@ folly::Future<arcticdb::entity::VariantKey> write_incomplete_frame(
     const StreamId& stream_id,
     const std::shared_ptr<InputTensorFrame>& frame,
     bool validate_index,
-    std::optional<AtomKey>&& next_key)  {
+    std::optional<AtomKey>&& next_key,
+    const std::optional<std::vector<std::string>>& sort_columns)  {
     using namespace arcticdb::pipelines;
 
     sorting::check<ErrorCode::E_UNSORTED_DATA>(
-            !validate_index || index_is_not_timeseries_or_is_sorted_ascending(*frame),
+            !validate_index || sort_columns || index_is_not_timeseries_or_is_sorted_ascending(*frame),
             "When writing/appending staged data in parallel, input data must be sorted.");
 
     auto index_range = frame->index_range;
     auto segment = incomplete_segment_from_frame(frame, 0, std::move(next_key), false);
-    return store->write(
-        KeyType::APPEND_DATA,
-        VersionId(0),
-        stream_id,
-        index_range.start_,
-        index_range.end_,
-        std::move(segment));
+
+    if(sort_columns && !sort_columns->empty()) {
+        auto mutable_seg = segment.clone();
```

Comment:
Why?

Reply:
The segment at this point is composed of external data blocks - it's pointing to numpy arrays owned by Python. Probably we should have a not in-place segment sort as per your comment, but I left that for another day.